src/ipa_pc/mod.rs

```diff
@@ -1202,146 +1220,204 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
+        lc_comm = lc_comm.iter().enumerate().map(|(index, &lc_comm_segment)| {
+            let h_comm_segment = if h_comm.len() > index {
+                h_comm[index]
+            } else {
+                G::zero()
+            };
+            (lc_comm_segment.into_projective() + &h_comm_segment.mul(-z_x_value)).into_affine()
+        }).collect::<Vec<_>>();
+
+        if has_hiding {
+            for (index, &rand_segment) in h_randomness.rand.iter().enumerate() {
+                if index == lc_randomness.rand.len() {
+                    lc_randomness.rand.push(G::ScalarField::zero());
+                }
+                lc_randomness.rand[index] += rand_segment * -z_x_value;
+            }
         }
 
-        // h(X) polynomial added to the set of polynomials for multi-poly single-point batching
-        let mut labeled_polynomials = labeled_polynomials;
+        // TODO: after optimization we have only one polynomial, so this is actually
+        //       single poly single point opening
+        //       should be taken into account during the refactoring
+
+        // LC(p_1(X),p_2(X),...,p_m(X),h(X)) polynomial added to the set of polynomials for multi-poly single-point batching
         let labeled_batch_polynomial = LabeledPolynomial::new(
-            format!("Batch"),
-            batch_polynomial,
+            format!("LC"),
+            lc_polynomial,
             None,
             if has_hiding { Some(1) } else { None }
         );
-        labeled_polynomials.push(&labeled_batch_polynomial);
+        let labeled_polynomials = vec![&labeled_batch_polynomial];
 
-        // Commitment of h(X) polynomial added to the set of polynomials for multi-poly single-point batching
-        let mut commitments = commitments;
+        // Commitment to LC(p_1(X),p_2(X),...,p_m(X),h(X)) polynomial added to the set of polynomials for multi-poly single-point batching
         let labeled_batch_commitment = LabeledCommitment::new(
-            format!("Batch"),
-            Commitment { comm: batch_commitment.clone(), shifted_comm: None },
+            format!("LC"),
+            Commitment { comm: lc_comm.clone(), shifted_comm: None },
             None
         );
-        commitments.push(&labeled_batch_commitment);
-
-        let mut rands = rands;
-        let labeled_batch_rand = LabeledRandomness::new(format!("Batch"), batch_randomness);
-        rands.push(&labeled_batch_rand);
+        let batch_commitments = vec![&labeled_batch_commitment]; // commitments;
 
-        // absorb the evaluations at the new challenge x
-        // The value of `batch_commitment` is determined by these and the initial 
-        // opening claims
-        fs_rng.absorb(&to_bytes![batch_values].unwrap());
+        let labeled_batch_rand = LabeledRandomness::new(format!("LC"), lc_randomness);
+        let rands = vec![&labeled_batch_rand];
 
         let proof = Self::open_individual_opening_challenges(
```

Comment:
Same comment as for the verifier, probably even worse since you already took care of hiding in this function, and that's something the open_individual_opening_challenges function already does